Packet++/header/LdapLayer.h

```diff
@@ -569,7 +570,210 @@ namespace pcpp
+			: LdapLayer(std::move(asn1Record), data, dataLen, prevLayer, packet) {}
+
 		std::string getExtendedInfoString() const override;
+	private:
+		static constexpr int versionIndex = 0;
+		static constexpr int nameIndex = 1;
+		static constexpr int credentialIndex = 2;
+
+		static constexpr int saslMechanismIndex = 0;
+		static constexpr int saslCredentialsIndex = 1;
+	};
+
+	/**
+	 * @class LdapBindResponseLayer
+	 * Represents LDAP bind response operation
+	 */
+	class LdapBindResponseLayer : public LdapResponseLayer
+	{
+	public:
+		/**
+		 * A constructor to create a new LDAP bind response message
+		 * @param[in] messageId The LDAP message ID
+		 * @param[in] resultCode The LDAP result code
+		 * @param[in] matchedDN The distinguished name (DN) to set on the message. If not applicable
+		 * pass an empty string
+		 * @param[in] diagnosticMessage The additional information to set on the message. If not applicable
+		 * pass an empty string
+		 * @param[in] referral A list of URIs to re-try the operation somewhere else. This is an optional
+		 * parameter. If not provided then referral won't be added to the message
+		 * @param[in] serverSaslCredentials Encoded server SASL credentials for use in subsequent processing
+		 * @param[in] controls A vector of LDAP controls. This is an optional parameter, if not provided the message
+		 * will be created without LDAP controls
+		 */
+		LdapBindResponseLayer(uint16_t messageId, const LdapResultCode& resultCode, const std::string& matchedDN,
+			const std::string& diagnosticMessage, const std::vector<std::string>& referral = std::vector<std::string>(),
+			const std::vector<uint8_t>& serverSaslCredentials = std::vector<uint8_t>(),
+			const std::vector<LdapControl>& controls = std::vector<LdapControl>());
+
+		/**
+		 * @return Encoded server SASL credentials for use in subsequent processing
+		 */
+		std::vector<uint8_t> getServerSaslCredentials();
+	protected:
+		friend LdapLayer* LdapLayer::parseLdapMessage(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet);
+
+		static constexpr int serverSaslCredentialsTagType = 7;
+
+		LdapBindResponseLayer(std::unique_ptr<Asn1Record> asn1Record, uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet)
+			: LdapResponseLayer(std::move(asn1Record), data, dataLen, prevLayer, packet) {}
+	};
+
+	/**
+	 * @class LdapUnbindRequestLayer
+	 * Represents LDAP unbind operation
+	 */
+	class LdapUnbindRequestLayer : public LdapLayer
+	{
+	public:
+		// cppcheck-suppress noExplicitConstructor
+		/**
+		 * A constructor to create a new LDAP unbind message
+		 * @param[in] messageId The LDAP message ID
+		 * @param[in] controls A vector of LDAP controls. This is an optional parameter, if not provided the message
+		 * will be created without LDAP controls
+		 */
+		LdapUnbindRequestLayer(uint16_t messageId, const std::vector<LdapControl>& controls = std::vector<LdapControl>());
```

Comment:
I'm not sure why `cppcheck` complains about this c'tor but not about the c'tors of the other messages 🤔

Reply:
Regardless, I added `explicit` in this commit: https://github.com/seladb/PcapPlusPlus/pull/1448/commits/57fa2ab30f23dc902d443633a3927974f89f6f06

Reply:
> I'm not sure why `cppcheck` complains about this c'tor but not about the c'tors of the other messages 🤔

Coz there is only 1 param without a default value, so it can be called as an implicit conversion constructor unless marked `explicit`.